modules/subscriptions/subscriptions/subscriptions.test.tsx

```diff
@@ -0,0 +1,197 @@
+
+import React, { act } from "react";
+import { getComponent } from "@/resources/tests/test.utils";
+import Service from "@/core/module/service.factory";
+import { jest } from "@jest/globals";
+
+jest.mock("@/core/module/module.manager", () => ({
+  ModuleManager: {
+    get: jest.fn().mockReturnValue({
+      components: {
+        services: jest.fn(),
+      },
+    }),
+  },
+}));
+
+describe("===== PAYMENTS =====", () => {
+  describe("Subscriptions Page", () => {
+    let Subscriptions: CoreComponent;
+
+    beforeAll(() => {
+      Subscriptions = getComponent("app", "subscriptions");
+    });
+
+    beforeEach(() => {
+      delete window.location;
+      window.location = { href: "" } as Location;
+      jest.clearAllMocks();
+    });
+
+    afterEach(() => {});
+
+    it("Templates is Mounted", async () => {
+      await act(() => render(<Subscriptions />));
+
+      expect(screen.getByText("subscriptions.title")).toBeInTheDocument();
+      expect(screen.getByText("subscriptions.sub-title")).toBeInTheDocument();
+      screen.debug();
+    });
+
+    it("Checks both pricing cards are displayed", async () => {
+      render(<Subscriptions />);
+
+      await waitFor(() => {
+        expect(
+          screen.getAllByTestId("subscriptions.pricing-card"),
+        ).toHaveLength(2);
+      });
+    });
+
+    it("Checks pricing card price is loaded", async () => {
+      render(<Subscriptions />);
+
+      await waitFor(() => {
+        expect(
+          screen.getAllByText(/subscriptions\.price-\d+\.currency/),
+        ).toHaveLength(2);
+      });
+    });
+
+    it("Checks all features have been loaded", async () => {
+      render(<Subscriptions />);
+
+      await waitFor(() => {
+        expect(
+          screen.getAllByText(
+            /subscriptions\.price-\d+\.features\.feature-\d+/,
+          ),
+        ).toHaveLength(3);
+      });
+
+      expect(
+        screen.getAllByText(/subscriptions\.price-1\.features\.feature-\d+/),
+      ).toHaveLength(1);
+      expect(
+        screen.getAllByText(/subscriptions\.price-2\.features\.feature-\d+/),
+      ).toHaveLength(2);
+    });
+
+    it("Checks subscribe button on first price card redirects", async () => {
+      render(<Subscriptions />);
+
+      const subscribeButton = await screen.findByText(
+        "subscriptions.price-1.btn-text",
+      );
+      await userEvent.click(subscribeButton);
+
+      expect(window.location.href).toBe(
+        "https://invoxmedical.typeform.com/invox-genesis?typeform-source=genesismedical.ai",
+      );
+    });
+
+    it("Checks subscribe button on second price card redirects", async () => {
+      const mockGetSubscriptionLink = jest
+        .fn<() => Promise<{ url: string }>>()
+        .mockResolvedValue({ url: "https://example.com/subscribe" });
+
+      const serviceGetSpy = jest.spyOn(Service, "get").mockReturnValue({
+        getSubscriptionLink: mockGetSubscriptionLink,
+      } as any);
```

Comment:
You can mock directly the service, as we have the instance `spyOn(subscriptionService, ' getSubscriptionLink`)  .mockOnceResolvedValue({ url: "https://example.com/subscribe" });` (Interesting the use of Once)

That will also remove the `any` here and simplify the block of code